src/routes/v1/friendships/history.rs

```diff
@@ -22,26 +24,36 @@ use crate::{
 use super::types::MessageRequestEvent;
 
 #[derive(Deserialize, Serialize)]
-pub struct RequestEventRequestBody {
+pub struct RequestEventParam {
     pub timestamp_from: i64, // timestamp in milis
     pub timestamp_to: i64,   // timestamp in milis
 }
 
 #[get("/v1/friendships/{friendshipId}/request-events/messages")]
 async fn get_sent_messages_request_event(
     req: HttpRequest,
-    body: web::Json<RequestEventRequestBody>,
+    param: web::Query<RequestEventParam>,
     friendship_id: web::Path<Uuid>,
     app_data: Data<AppComponents>,
 ) -> Result<HttpResponse, FriendshipsError> {
-    let _logged_in_user: UserId = req
+    let logged_in_user: UserId = req
         .extensions()
         .get::<UserId>()
         .expect("to have a UserId")
         .clone();
 
+    // Retrieve users from friendship and verify permissions.
+    let users = get_users_friendship(app_data.clone(), *friendship_id).await;
+    if !users.is_empty() {
+        has_permission(logged_in_user.social_id.as_str(), &users[0], &users[1]);
+    } else {
+        return Err(FriendshipsError::CommonError(CommonError::BadRequest(
+            format!("You don't have permission to view the sent request messages for friendship {friendship_id}"),
```

Comment:
is this error description ok? I mean you are returning that the client doesn't have permission but the `if else` statement is checking that the `users` `Vec` is not empty.

Reply:
nice catch, see suggestion below

Reply:
Hahah nope, does not make sense at all. great catch ✨